forem/forem
# Adds validations to avoid creating unnecessary PNs

## What type of PR is this? (check all applicable)


## Description

When the default `com.forem.app` ConsumerApp doesn't have its credentials (any selfhost instance) and someone uses the mobile app an error was surfacing. This PR fixes this problem.

## Related Tickets & Documents

Closes #14613

## QA Instructions, Screenshots, Recordings

1. Boot app locally
2. Login (register new account if needed) using iOS mobile app
   - If you don't have an iOS device you can use your usual `localhost:3000`. Open up a rails console and create a dummy `Device` associated to the user and the default `ConsumerApp` to simulate the PN registration
   - If you are using the iOS app you'll need to use ngrok for HTTPS support
3. Write a comment on any Post
4. Use a different account to reply to that ^ comment
5. You shouldn't see the error mentioned in #14613 

Let me know if you have any problems testing this, I'd be happy to help with either ngrok + iOS app or just plain `localhost:3000` option.

### UI accessibility concerns?

None - backend related change

## Added/updated tests?


## [Forem core team only] How will this change be communicated?


This is a small bug fix that solves a problem that comes up on 

## [optional] What gif best describes this PR or how it makes you feel?

![The Simpsons certificate: Not insane](https://media.giphy.com/media/3orieJZZd34v3zkXFS/giphy.gif?cid=ecf05e47dk3o2y37ifx9b1wo85t3i16prk6hr3idij0x91p9&rid=giphy.gif&ct=g)
